transport_plugins/bled112/iotile_transport_bled112/hardware/async_central.py

```diff
@@ -0,0 +1,260 @@
+        for conn in active_conns:
+            await self._bled112.disconnect(conn.conn_handle)
+
+        await self._bled112.stop()
+
+    # async def connect(self, conn_string: str, delegate: Optional[BLEPeripheralDelegate] = None) -> BLEPeripheral:
+    #     """
+    #     """
+
+    #     start_time = time.monotonic()
+
+    #     handle = await self._bled112.connect(conn_string)
+
+    #     try:
+    #         table = await self._bled112.probe_gatt_table(handle)
+    #     except errors.DisconnectionError:
+    #         raise
+    #     except errors.LinkError:
+    #         try:
+    #             await self._bled112.disconnect(handle)
+    #         except:  #pylint:disable=bare-except;We want to raise the original error; this one is logged.
+    #             self._logger.exception("Error disconnecting from device during failed gatt table probe")
+
+    #         raise
+
+    #     end_time = time.monotonic()
+    #     self._logger.info("Total time to connect to device %.3f", end_time - start_time)
+
+    #     peripheral = BLED112Peripheral(conn_string, handle, table)
+    #     if delegate is None:
+    #         delegate = EmptyPeripheralDelegate()
+
+    #     data = _ConnectionData(conn_string, handle, delegate, peripheral)
+    #     self._tracked_conns[handle] = data
+
+    #     return peripheral
+
+    async def disconnect(self, peripheral: BLEPeripheral):
+        """
+        """
+
+        data = self._ensure_connected(peripheral)
+
+        # Connection is cleaned up by disconnect handler
+        await self._bled112.disconnect(data.conn_handle)
+        assert data.conn_handle not in self._tracked_conns
+
+    async def set_notifications(self, peripheral: BLEPeripheral, characteristic: UUID,
+                                enabled: bool, kind: str = 'notify'):
+        """
+        """
+
+        if kind not in ('notify', 'indicate'):
+            raise errors.UnsupportedOperationError("Invalid notification type: %s" % kind)
+
+        if kind != 'notify':
+            raise errors.UnsupportedOperationError("Cannot configure notification type %s" % kind)
+
+        data = self._ensure_connected(peripheral)
+        await self._bled112.set_subscription(data.conn_handle, peripheral.find_char(characteristic),
+                                             kind, enabled)
+
+    async def write(self, peripheral: BLEPeripheral, characteristic: UUID,
+                    value: bytes, with_response: bool):
+        """
+        """
+
+        data = self._ensure_connected(peripheral)
+        handle = peripheral.prepare_write(characteristic, value)
+        await self._bled112.write_handle(data.conn_handle, handle, with_response, value)
+
+    async def read(self, peripheral: BLEPeripheral, characteristic: UUID):
+        """
+        """
+
+        #FIXME: Raise an exception here
+
+    async def request_scan(self, tag: str, active: bool, delegate: BLEScanDelegate = None):
```

Comment:
@DimaWittmann The BLEScanDelegate concept I believe is old and can be removed.  Previously scan events came via callbacks made on the BLEScanDelegate object passed in but we moved to centralizing that through the `events` in the async central so this shouldn't be needed anymore and can just be removed completely.